k4FWCore/components/CollectionMerger.cpp

```diff
@@ -69,20 +69,20 @@ struct CollectionMerger final : k4FWCore::Transformer<std::shared_ptr<podio::Col
     m_map["edm4hep::VertexCollection"]        = &CollectionMerger::mergeCollections<edm4hep::VertexCollection>;
     m_map["edm4hep::ReconstructedParticleCollection"] =
         &CollectionMerger::mergeCollections<edm4hep::ReconstructedParticleCollection>;
-    m_map["edm4hep::MCRecoParticleAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoParticleAssociationCollection>;
-    m_map["edm4hep::MCRecoCaloAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoCaloAssociationCollection>;
-    m_map["edm4hep::MCRecoTrackerAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoTrackerAssociationCollection>;
-    m_map["edm4hep::MCRecoCaloParticleAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoCaloParticleAssociationCollection>;
-    m_map["edm4hep::MCRecoClusterParticleAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoClusterParticleAssociationCollection>;
-    m_map["edm4hep::MCRecoTrackParticleAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::MCRecoTrackParticleAssociationCollection>;
-    m_map["edm4hep::RecoParticleVertexAssociationCollection"] =
-        &CollectionMerger::mergeCollections<edm4hep::RecoParticleVertexAssociationCollection>;
+    m_map["edm4hep::RecoMCParticleLinkCollection"] =
```

Comment:
Here one could think that duplicating the lines to also have the old names in the map (mapped to merging the new ones) could be useful but files with the old names can have been produced with the latest release and are incompatible or with an older nightly which will also be incompatible.

Reply:
In principle the new typenames should be enough because they come from the collection in any case. So even if an older version / name was on disk, by the time it reaches this place, we will only ever have to deal with the actual / new typename.